OQ_Toolkit/OQ_ARVRController/scripts/Feature_RigidBodyGrab.gd

```diff
@@ -1,65 +1,56 @@
 extends Spatial
 
 var controller : ARVRController = null;
-
 var grab_area : Area = null;
 var held_object = null;
 var held_object_data = {};
-
 var grab_mesh : MeshInstance = null;
-export var reparent_mesh = false;
-
-onready var grabbed_object_script = preload("helper_grabbed_RigidBody.gd");
-
 
 enum {
 	GRABTYPE_VELOCITY,
 	GRABTYPE_PINJOINT, #!!TODO: not yet working; I first need to figure out how joints work
 }
-
 var grab_type = GRABTYPE_VELOCITY;
 
+onready var main_node = get_node("/root/Main")
+
+export var reparent_mesh = false;
+
 func _ready():
 	controller = get_parent();
 	if (not controller is ARVRController):
 		vr.log_error(" in Feature_RigidBodyGrab: parent not ARVRController.");
 	grab_area = $GrabArea;
+	
+	controller.connect("button_pressed", self, "_on_ARVRController_button_pressed")
+	controller.connect("button_release", self, "_on_ARVRController_button_release")
 
 
-func start_grab_velocity(rigid_body):
-	if (rigid_body.get_script() == grabbed_object_script):
-		print("Double grab... not yet supported for velocity grab");
-	else:
-		held_object = rigid_body;
-		held_object_data["script"] = held_object.get_script();
-		held_object.set_script(grabbed_object_script);
-		held_object.grab_init(self);
-		if (reparent_mesh):
```

Comment:
Where is the code for when reparent_mesh is set to true? I think this is a very nice feature for less perceived latency when holding rigid body objects and we should not loose it.

Reply:
Good catch, deleted it accidently.